sources/iTermColorMap.m

```diff
@@ -86,19 +87,30 @@ - (void)setMutingAmount:(double)mutingAmount {
 }
 
 - (void)setColor:(NSColor *)theColor forKey:(iTermColorMapKey)theKey {
-    if (!theColor || theColor == _map[@(theKey)] || theKey >= kColorMap24bitBase) {
+    if (theKey >= kColorMap24bitBase)
+        return;
+
+    if ( !theColor ) {
+        [_map removeObjectForKey:@(theKey)];
```

Comment:
The change to remove the object from the map when the color is nil will change behavior when called from `-[PTYSession screenSetColor:forKey:]`. This would happen when we get an escape sequence to change the color map with an improperly specified argument. It used to have no effect, but now it would leave the colormap with a missing entry. Guard against it by changing `-[PTYSession screenSetColor:forKey:]` to do nothing if its color argument is `nil`.